eng/common/TestResources/Remove-TestResources.ps1

```diff
@@ -138,8 +138,8 @@ if (![string]::IsNullOrWhiteSpace($ServiceDirectory)) {
 }
 
 Log "Deleting resource group '$ResourceGroupName'"
-if (Retry { Remove-AzResourceGroup -Name "$ResourceGroupName" -Force:$Force }) {
-    Write-Verbose "Successfully deleted resource group '$ResourceGroupName'"
+if (Remove-AzResourceGroup -Name "$ResourceGroupName" -Force:$Force -AsJob) {
```

Comment:
Is this the correct option? Wouldn't this fail most of the time as soon as the pwsh process is closed? Or would it block the PS process from shutting down? 

I was hoping there was some async call we could make on the azure side to start this process.

Reply:
You and me both @weshaggard! I was surprised, because the python azure cli has `--no-wait`, but the powershell cmdlets only have `AsJob`, so I was frustrated by the inconsistency.

That said, I was worried about this same thing. In my testing I found that the below code was sufficient enough to trigger a resource group deletion. I'll do a little more investigation though, as I'd like to know whether we're just getting lucky with timing, or if it only delegates the async polling to the background job.

```
Remove-AzResourceGroup -Name <rg> -Force -AsJob ; Exit
```

What I'm removing is a retry on some 4xx/5xx failures, but I figured it would be sufficient to have the background poller remove those. That said, if we see some persistent ARM issues, it could cause a lot of resources to pile up, though I would argue that's a better case than blocking the pipeline timing anyway (and as @mikeharder suggested somewhere we may want to bump the deletion poller frequency anyway).

Reply:
In practice it might not matter, but to be safe I think we should block until `Get-AzResourceGroup` returns `ProvisioningState: Deleting`.  This is very fast (usually less than one second) and confirms the service has received the request.  Here's a timeline:

```
> New-AzResourceGroup -name foo -location westus2

> Get-AzResourceGroup -name foo
ProvisioningState : Succeeded

> Remove-AzResourceGroup -Name foo -Force -AsJob

> Get-AzResourceGroup -name foo
ProvisioningState : Deleting
```

Reply:
Updated. Please check out the latest changes.